### DISCUSSION TITLE
Output triggered by falling edge successively

### DISCUSSION THREAD

**Discussion by goyukiya** [bonsai-rx/bonsai#1729] [Q&A] [answered]
hi, 

i am currently trying to make a workflow to send a TTL output of 1 sec. when a falling edge is detected on the input channel.

the worklow works fine in most cases but produces 2 unwanted behaviors:
1. an output is generated everytime when at the start of the worklow. is there a way to prevent that?
2. when a trigger is detected while there is an ongoing output, the output seems to be cut short and triggered again by the second trigger.  i would like to wait for the output to finish before accepting any new trigger. 

i am new to bonsai so please let me know if there is anything wrong.
thanks,
yuki

![demo](https://github.com/bonsai-rx/bonsai/assets/46738403/2d2bf3a9-2f1f-4e55-91b5-eef82deb81a9)

Edit:
attached the workflow
[triggerOut.zip](https://github.com/bonsai-rx/bonsai/files/14925726/triggerOut.zip)

**Comment by bruno-f-cruz** [comment]
Include the workflow, thanks!

**Reply by goyukiya** [reply]
I edited the post to include the workflow, thanks!

**Reply by bruno-f-cruz** [reply]
I refactored the workflow with some advised better practices. Regarding your specific points:

1. You can add a delay subscription to the whole branch to delay its start.
2. You can close the sequence when one trigger is received and use repeat to re-arm
I dont have an arduino to test so let me know if something doesnt work as intended.

good luck!
[triggerOut_refactored.zip](https://github.com/bonsai-rx/bonsai/files/14926881/triggerOut_refactored.zip)

**Reply by goyukiya** [reply]
thanks for the help!
(2) seems to be fixed. for my understanding, is it the combination of Take (1) and Repeat at the end that closes the sequence and rearms it?

for (1), the delay subscription only adds a delay but does not prevent the first output at startup. i feel it is due to the fact that the input channel is always LOW at startup in my setup. is there a way to delay the start of the sequence until the input is HIGH?

**Reply by bruno-f-cruz** [reply]
1) I am a bit confused. What exactly do you want to do? What should be the initial state of the output pin when the task starts? What should happen if the initial state of the input is a) Low and b) High? You should be able to define all these states...
2) yes

**Reply by goyukiya** [reply]
sorry for the confusion. i forgot to mention these elements.
i am listening to TTL pulses on the input port and the input is always low at startup. Once a TTL is received, i want to generate the output only on the falling edge of the TTL, nothing happening on the rising edge. the output is low at startup.

should i try to detect the rising edge first and then wait for the falling edge to continue the sequence? or is there a more efficient way to proceed?

thanks for your help!

**Reply by bruno-f-cruz** [reply, marked as answer]
I think simply subscribing to the sequence when you have the first rising edge should do the trick. 
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/a69120db-94ad-4f36-b12e-26f330a58fc1)

**Reply by goyukiya** [reply]
Great! that worked!

Thanks a lot!
